### PR TITLE
714 push requisitions 2

### PR DESCRIPTION
**Pull request by clemens-msupply**

### CODE REVIEW

**Review by andreievg — APPROVED** [review]
Looks good thanks, although some change requests, and a couple of general comments:

I think we should try to make our translations as simple as possible, it's definitely an area that can cause a lot of confusion (especially when someone else looks at it or when looking at it when it's not fresh in ones memory). On that note:

* Full mSupply schema is very vast and quite complex, and we don't need to translate everything, just the bits we are interested in for now (like I don't really see any point knowing all variants of invoice etc.., I think it just clutters without too much benefit).
* As perviously pointed out, I really like serde rename pattern for simple translation, especially now that we are renaming in two different places (push and pull), i.e.: https://github.com/openmsupply/remote-server/compare/714-push-requisitions-2...Rename-example  (not suggesting we refactor just now, but thought to point it out)
* Do you want me to do a full list of fields that need to be added to mSupply ? Or was that on your todo ?

File: server/src/sync/translation_remote/requisition.rs [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+fn to_legacy_status(status: &RequisitionRowStatus) -> LegacyRequisitionStatus {
+    match status {
+        RequisitionRowStatus::New => LegacyRequisitionStatus::Sg,
+        RequisitionRowStatus::Draft => LegacyRequisitionStatus::Sg,
+        RequisitionRowStatus::Sent => LegacyRequisitionStatus::Cn,
+        RequisitionRowStatus::Finalised => LegacyRequisitionStatus::Fn,
+    }
+}
+
+fn hex_colour_to_req_colour(colour: &str) -> Option<i64> {
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply server/src/sync/translation_remote/requisition.rs:268]: yes sounds good will remove once we have the missing fields in mSupply

File: server/src/sync/translation_remote/requisition.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+fn to_legacy_type(t: &RequisitionRowType) -> LegacyRequisitionType {
+    match t {
+        RequisitionRowType::Request => LegacyRequisitionType::Request,
+        RequisitionRowType::Response => LegacyRequisitionType::Response,
+    }
+}
+
+fn from_legacy_status(status: &LegacyRequisitionStatus) -> Option<RequisitionRowStatus> {
+    let status = match status {
+        LegacyRequisitionStatus::Sg => RequisitionRowStatus::New,
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply server/src/sync/translation_remote/requisition.rs]: thanks
[71be2cd](https://github.com/openmsupply/remote-server/pull/886/commits/71be2cd5b3ae0150e7c2405ed36ca7fe851d1551)

File: server/src/sync/translation_remote/requisition.rs [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+                    .date_order_received
+                    .map(|date| date_and_time_to_datatime(date, 0)),
+                // TODO needs new field in mSupply
+                finalised_datetime: None,
+                colour: data.colour.and_then(|colour| req_colour_to_hex(colour)),
+                comment: data.comment,
+                // TODO correct?:
+                their_reference: data.requester_reference,
+                // TODO:
+                max_months_of_stock: 0.0,
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply server/src/sync/translation_remote/requisition.rs]: [7d98c9d](https://github.com/openmsupply/remote-server/pull/886/commits/7d98c9dd6e599b316cab895428147fd6f14b5785)

**Comment by clemens-msupply** [comment]
thanks for your comments! plan to compose a list of all missing mSupply fields

**Pull request merged**